tests/Unit/Core/FeatureFlag/FeatureFlagsModifierTest.php

```diff
@@ -0,0 +1,258 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Unit\Core\FeatureFlag;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\ObjectRepository;
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+use PrestaShop\PrestaShop\Core\FeatureFlag\FeatureFlagsModifier;
+use PrestaShopBundle\Entity\FeatureFlag;
+use Symfony\Component\Translation\TranslatorInterface;
+use TypeError;
+
+class FeatureFlagsModifierTest extends TestCase
+{
+    public function testImplementsDataConfigurationInterface()
+    {
+        $entityManagerMock = $this->getMockBuilder(EntityManagerInterface::class)->getMock();
+        $translatorMock = $this->getMockBuilder(TranslatorInterface::class)->getMock();
+        $modifier = new FeatureFlagsModifier($entityManagerMock, $translatorMock);
+
+        $this->assertTrue($modifier instanceof DataConfigurationInterface);
+    }
+
+    public function testGetConfiguration()
+    {
+        $featureFlags = [
+            new FeatureFlag('product_page_v1'),
+            (new FeatureFlag('product_page_v2'))->enable(),
+            new FeatureFlag('product_page_v3'),
+        ];
+
+        list($entityManagerMock, $repositoryMock) = $this->buildDoctrineServicesMocksForFetchAll($featureFlags);
+        $translatorMock = $this->getMockBuilder(TranslatorInterface::class)->getMock();
+
+        $modifier = new FeatureFlagsModifier($entityManagerMock, $translatorMock);
+
+        $expected = [
+            'product_page_v1' => false,
+            'product_page_v2' => true,
+            'product_page_v3' => false,
+        ];
+
+        $this->assertEquals($expected, $modifier->getConfiguration());
+    }
+
+    public function testGetEmptyConfiguration()
```

Comment:
```suggestion
    public function testGetConfigurationReturnsEmptyIfNoFeatureFlagsAvailable()
```

You can also clarify the intent of the test on a comment